src/aleph_pytezos/crypto/key.py

```diff
@@ -126,12 +129,17 @@ def from_secret_exponent(
         """
         # Ed25519
         if curve == b'ed':
-            raise Exception("Curve ed25519 not supported.")
             # Dealing with secret exponent or seed?
-            #if len(secret_exponent) == 64:
-            #    public_point = pysodium.crypto_sign_sk_to_pk(sk=secret_exponent)
-            #else:
-            #    public_point, secret_exponent = pysodium.crypto_sign_seed_keypair(seed=secret_exponent)
+            if len(secret_exponent) == 64:
+                # The standard format of a secret key is 64 bytes,
+                # the first 32 bytes are the public key, the last 32 are secret (or seed).
+                public_key, secret_exponent = secret_exponent[:32], secret_exponent[32:]
+                public_point = nacl.signing.SigningKey(secret_exponent).verify_key.encode()
+            elif len(secret_exponent) == 32:
+                keypair = nacl.signing.SigningKey(secret_exponent)
+                public_point = keypair.verify_key.encode()
+            else:
+                raise AssertionError()
```

Comment:
That is not a very Pythonic way of writing it, the following would be more clear.

```suggestion
            else:
                assert len(secret_exponent) == 32, f"Secret exponent length must be 32 or 64, not {len(secret_exponent)}"
                keypair = nacl.signing.SigningKey(secret_exponent)
                public_point = keypair.verify_key.encode()
```